From f5b7911d109999331468286a3bf09424e2d78d95 Mon Sep 17 00:00:00 2001
From: aconway <aconway@13f79535-47bb-0310-9956-ffa450edef68>
Date: Wed, 27 Jun 2012 20:34:57 +0000
Subject: [PATCH] NO-JIRA: Make qmf/BrokerImportExport.h header public

QMF code generator templates include this header so it is required to build QMF
generated code against the -devel packages.

Review: https://reviews.apache.org/r/5593/

git-svn-id: https://svn.apache.org/repos/asf/qpid/trunk@1354712 13f79535-47bb-0310-9956-ffa450edef68
---
 qpid/cpp/include/qmf/BrokerImportExport.h |   42 +++++++++++++++++++++++++++++
 qpid/cpp/src/CMakeLists.txt               |    1 +
 qpid/cpp/src/qmf.mk                       |    4 +--
 qpid/cpp/src/qmf/BrokerImportExport.h     |   42 -----------------------------
 4 files changed, 45 insertions(+), 44 deletions(-)
 create mode 100644 qpid/cpp/include/qmf/BrokerImportExport.h
 delete mode 100644 qpid/cpp/src/qmf/BrokerImportExport.h

diff --git a/qpid/cpp/include/qmf/BrokerImportExport.h b/qpid/cpp/include/qmf/BrokerImportExport.h
new file mode 100644
index 0000000..ee05788
--- /dev/null
+++ b/qpid/cpp/include/qmf/BrokerImportExport.h
@@ -0,0 +1,42 @@
+#ifndef QPID_BROKER_IMPORT_EXPORT_H
+#define QPID_BROKER_IMPORT_EXPORT_H
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#if defined(WIN32) && !defined(QPID_DECLARE_STATIC)
+#  if defined(BROKER_EXPORT) || defined (qpidbroker_EXPORTS)
+#    define QPID_BROKER_EXTERN __declspec(dllexport)
+#  else
+#    define QPID_BROKER_EXTERN __declspec(dllimport)
+#  endif
+#  ifdef _MSC_VER
+#    define QPID_BROKER_CLASS_EXTERN
+#    define QPID_BROKER_INLINE_EXTERN QPID_BROKER_EXTERN
+#  else
+#    define QPID_BROKER_CLASS_EXTERN QPID_BROKER_EXTERN
+#    define QPID_BROKER_INLINE_EXTERN
+#  endif
+#else
+#  define QPID_BROKER_EXTERN
+#  define QPID_BROKER_CLASS_EXTERN
+#  define QPID_BROKER_INLINE_EXTERN
+#endif
+
+#endif
diff --git a/qpid/cpp/src/CMakeLists.txt b/qpid/cpp/src/CMakeLists.txt
index cc882f8..fff2a85 100644
--- a/qpid/cpp/src/CMakeLists.txt
+++ b/qpid/cpp/src/CMakeLists.txt
@@ -1190,6 +1190,7 @@ set (qmf_SOURCES
 set (qmf_HEADERS
   ../include/qpid/agent/ManagementAgent.h
   ../include/qpid/agent/QmfAgentImportExport.h
+  ../include/qmf/BrokerImportExport.h
     )
 
 add_msvc_version (qmf library dll)
diff --git a/qpid/cpp/src/qmf.mk b/qpid/cpp/src/qmf.mk
index 9b5df6c..6a4bce4 100644
--- a/qpid/cpp/src/qmf.mk
+++ b/qpid/cpp/src/qmf.mk
@@ -30,7 +30,8 @@ lib_LTLIBRARIES +=	\
 #
 QMF_API =					\
   ../include/qpid/agent/ManagementAgent.h	\
-  ../include/qpid/agent/QmfAgentImportExport.h
+  ../include/qpid/agent/QmfAgentImportExport.h	\
+  ../include/qmf/BrokerImportExport.h
 
 #
 # Public headers for the QMF2 API
@@ -96,7 +97,6 @@ libqmf2_la_SOURCES = 		\
   qmf/AgentSessionImpl.h	\
   qmf/AgentSubscription.cpp	\
   qmf/AgentSubscription.h	\
-  qmf/BrokerImportExport.h	\
   qmf/ConsoleEvent.cpp		\
   qmf/ConsoleEventImpl.h	\
   qmf/ConsoleSession.cpp	\
diff --git a/qpid/cpp/src/qmf/BrokerImportExport.h b/qpid/cpp/src/qmf/BrokerImportExport.h
deleted file mode 100644
index ee05788..0000000
--- a/qpid/cpp/src/qmf/BrokerImportExport.h
+++ /dev/null
@@ -1,42 +0,0 @@
-#ifndef QPID_BROKER_IMPORT_EXPORT_H
-#define QPID_BROKER_IMPORT_EXPORT_H
-
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-#if defined(WIN32) && !defined(QPID_DECLARE_STATIC)
-#  if defined(BROKER_EXPORT) || defined (qpidbroker_EXPORTS)
-#    define QPID_BROKER_EXTERN __declspec(dllexport)
-#  else
-#    define QPID_BROKER_EXTERN __declspec(dllimport)
-#  endif
-#  ifdef _MSC_VER
-#    define QPID_BROKER_CLASS_EXTERN
-#    define QPID_BROKER_INLINE_EXTERN QPID_BROKER_EXTERN
-#  else
-#    define QPID_BROKER_CLASS_EXTERN QPID_BROKER_EXTERN
-#    define QPID_BROKER_INLINE_EXTERN
-#  endif
-#else
-#  define QPID_BROKER_EXTERN
-#  define QPID_BROKER_CLASS_EXTERN
-#  define QPID_BROKER_INLINE_EXTERN
-#endif
-
-#endif
-- 
1.7.10.2

